legend-engine-xt-serviceStore-executionPlan/src/main/java/org/finos/legend/engine/plan/execution/stores/service/auth/ServiceStoreConnectionProvider.java

```diff
@@ -0,0 +1,108 @@
+import org.finos.legend.engine.protocol.pure.v1.model.packageableElement.store.service.model.SecurityScheme;
+import org.finos.legend.engine.shared.core.function.Function5;
+import org.finos.legend.engine.shared.core.identity.Credential;
+import org.finos.legend.engine.shared.core.identity.Identity;
+
+import java.util.List;
+import java.util.Objects;
+
+public class ServiceStoreConnectionProvider extends ConnectionProvider<HttpConnectionBuilder>
+{
+    public ServiceStoreConnectionProvider(CredentialProviderProvider credentialProviderProvider)
+    {
+        super(credentialProviderProvider);
+    }
+
+    public HttpConnectionBuilder makeConnection(ConnectionSpecification connectionSpecification, AuthenticationSpecification authenticationSpecification, Identity identity) throws Exception
+    {
+        if (!(connectionSpecification instanceof ServiceStoreConnectionSpecification && authenticationSpecification instanceof ServiceStoreAuthenticationSpecification))
+        {
+            throw new IllegalStateException("Invalid ConnectionSpecification/AuthenticationSpecification. Please reach out to dev team");
+        }
+
+        ServiceStoreConnectionSpecification serviceStoreConnectionSpecification = (ServiceStoreConnectionSpecification) connectionSpecification;
+        ServiceStoreAuthenticationSpecification serviceStoreAuthenticationSpecification = (ServiceStoreAuthenticationSpecification) authenticationSpecification;
+
+        if (serviceStoreAuthenticationSpecification.authenticationSchemeRequirements != null)
+        {
+            for (AuthenticationSchemeRequirement requirement : serviceStoreAuthenticationSpecification.authenticationSchemeRequirements)
+            {
+                SingleAuthenticationSchemeRequirement authenticationSchemeRequirement = (SingleAuthenticationSchemeRequirement) requirement;
+                SecurityScheme securityScheme = authenticationSchemeRequirement.securityScheme;
+                AuthenticationSpecification authSpecification = authenticationSchemeRequirement.authenticationSpecification;
+
+                HttpClientBuilder clientBuilder = HttpClients.custom();
+                RequestBuilder requestBuilder = makeRequestUtil(serviceStoreConnectionSpecification);
+                HttpConnectionBuilder httpConnectionBuilder = new HttpConnectionBuilder(clientBuilder, requestBuilder);
+                Credential credential = null;
+                if (authSpecification != null)
+                {
+                    credential = makeCredential(authSpecification, identity);
+                }
+                configureAuthentication(httpConnectionBuilder, securityScheme, credential, identity);
+                return httpConnectionBuilder;
+            }
+        }
+
+        return new HttpConnectionBuilder(HttpClients.custom(), makeRequestUtil(serviceStoreConnectionSpecification));
```

Comment:
this might break existing plans

Reply:
Fix in ServiceStoreAuthenticationSpecification constructor.

Reply:
thanks